common/thorhelper/roxiehelper.cpp

```diff
@@ -1648,6 +1646,31 @@ int readHttpHeaderLine(IBufferedSocket *linereader, char *headerline, unsigned m
     return bytesread;
 }
 
+void parseHttpParameterString(IProperties *p, const char *str)
+{
+    while (*str)
+    {
+        StringBuffer s, prop, val;
+        while (*str && *str != '&' && *str != '=')
+            s.append(*str++);
+        appendDecodedURL(prop, s.trim());
+        if (!*str || *str == '&')
+            val.set("1");
+        else
+        {
+            s.clear();
+            str++;
+            while (*str && *str != '&')
+                s.append(*str++);
+            appendDecodedURL(val, s.trim());
+        }
+        if (prop.length())
+            p->setProp(prop, val);
+        if (*str)
+            str++;
+    }
+}
```

Comment:
It's not likely to happen.  I assume the typical case for those is clicking on a link to another position in the same page?  But these are REST queries as a FORM POST or GET URL.  To some degree we can dictate what the format we support is, but I suppose we could treat the '#' as a terminator.  I assume nothing can follow '#' except the marker?